compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -10513,16 +10510,73 @@ OMR::Z::TreeEvaluator::arraycmpHelper(TR::Node *node,
             *returnCond = TR::InstOpCode::COND_MASK0;
          }
       }
+
    else if (isConstLength)
       {
+      TR::LabelSymbol * cFlowRegionEnd;
       if (node->getOpCode().hasSymbolReference() && node->getSymbolReference())
          {
          baseSource1Ref = generateS390MemoryReference(cg, node, source1Node, 0, true); // forSS=true
          baseSource2Ref = generateS390MemoryReference(cg, node, source2Node, 0, true); // forSS=true
+         if (baseSource1Ref->getOffset() == baseSource2Ref->getOffset())
+            {
+            // since forSS was true in the generation of the memory references, there will not be any index registers.
+            // if there are base registers, we can compare those and conclude if are equal.
+            if(baseSource1Ref->getBaseRegister() && baseSource2Ref->getBaseRegister())
```

Comment:
Yes, there should be base register (We have D(B) format or D(X,B) format memory reference). So there should always be. 

Though, can you elaborate the change here bit more?

Reply:
I thought it is possible to have only displacement in ss structure but I was wrong. I think it is safe to assume the index register is null here since the `forSS` flag is set. first I check to make sure the displacements are equal and if it is, I compare base registers. But It seems that when we create a memRef from node, the `_baseRegister` property is null.